compiler/rustc_interface/src/passes.rs

```diff
@@ -362,7 +362,7 @@ pub fn configure_and_expand(
         if recursion_limit_hit {
             // If we hit a recursion limit, exit early to avoid later passes getting overwhelmed
             // with a large AST
-            Err(ErrorReported)
+            Err(ErrorReported::new_i_know_what_im_doing())
```

Comment:
Looks like `reduced_recursion_limit` could contain an `ErrorReported`.

Reply:
Coming from where?

Reply:
From whatever set it, which presumably produced an error in the process.

Reply:
IIUC, it starts out set and only becomes `None` if there is no error... so there may not be another good place to generate an `ErrorReported`...